models/test-management/src/types.ts

```diff
@@ -0,0 +1,221 @@
+  Prop,
+  TypeRef,
+  UX,
+  TypeMarkup,
+  Index,
+  TypeCollaborativeDoc,
+  TypeString,
+  Collection,
+  ReadOnly,
+  TypeNumber,
+  TypeDate
+} from '@hcengineering/model'
+import attachment from '@hcengineering/model-attachment'
+import core, { TAttachedDoc, TDoc, TType, TTypedSpace } from '@hcengineering/model-core'
+
+import testManagement from './plugin'
+
+export { testManagementId } from '@hcengineering/test-management/src/index'
+
+export const DOMAIN_TEST_MANAGEMENT = 'test-management' as Domain
+
+/** @public */
+export function TypeTestCaseType (): Type<TestCaseType> {
+  return { _class: testManagement.class.TypeTestCaseType, label: testManagement.string.TestCaseType }
+}
+
+@Model(testManagement.class.TypeTestCaseType, core.class.Type, DOMAIN_TEST_MANAGEMENT)
+@UX(testManagement.string.TestCaseType)
+export class TTypeTestCaseType extends TType {}
+
+/** @public */
+export function TypeTestCasePriority (): Type<TestCasePriority> {
+  return { _class: testManagement.class.TypeTestCasePriority, label: testManagement.string.TestCasePriority }
+}
+
+@Model(testManagement.class.TypeTestCasePriority, core.class.Type, DOMAIN_TEST_MANAGEMENT)
+@UX(testManagement.string.TestCasePriority)
+export class TTypeTestCasePriority extends TType {}
+
+/** @public */
+export function TypeTestCaseStatus (): Type<TestCaseStatus> {
+  return { _class: testManagement.class.TypeTestCaseStatus, label: testManagement.string.TestCaseStatus }
+}
+
+@Model(testManagement.class.TypeTestCaseStatus, core.class.Type, DOMAIN_TEST_MANAGEMENT)
+@UX(testManagement.string.TestCaseStatus)
+export class TTypeTestCaseStatus extends TType {}
+
+@Model(testManagement.class.TestProject, core.class.TypedSpace)
+@UX(testManagement.string.TestProject)
+export class TTestProject extends TTypedSpace implements TestProject {
+  @Prop(TypeMarkup(), testManagement.string.FullDescription)
+  @Index(IndexKind.FullText)
+    fullDescription?: string
+}
+
+@Mixin(testManagement.mixin.DefaultProjectTypeData, testManagement.class.TestProject)
+@UX(getEmbeddedLabel('Default project'), testManagement.icon.TestProject)
+export class TDefaultProjectTypeData extends TTestProject implements RolesAssignment {
+  [key: Ref<Role>]: Ref<Account>[]
+}
+
+/**
+ * @public
+ */
+@Model(testManagement.class.TestSuite, core.class.Doc, DOMAIN_TEST_MANAGEMENT)
+@UX(testManagement.string.TestSuite, testManagement.icon.TestSuite, testManagement.string.TestSuite)
+export class TTestSuite extends TDoc implements TestSuite {
+  @Prop(TypeString(), testManagement.string.SuiteName)
+  @Index(IndexKind.FullText)
+    name!: string
+
+  @Prop(TypeMarkup(), testManagement.string.SuiteDescription)
+  @Index(IndexKind.FullText)
+    description?: string
+
+  @Prop(TypeRef(testManagement.class.TestSuite), testManagement.string.TestSuite)
+    parent?: Ref<TestSuite>
+
+  declare space: Ref<TestProject>
+}
+
+/**
+ * @public
+ */
+@Model(testManagement.class.TestCase, core.class.Doc, DOMAIN_TEST_MANAGEMENT)
```

Comment:
You should use `core.class.AttachedDoc` as parent class for classes that extend `AttachedDoc`

Reply:
Thank you, fixed